frontend/www/js/omegaup/user/profile.ts

```diff
@@ -78,6 +80,16 @@ OmegaUp.on('ready', () => {
           ) => {
             api.User.update(userBasicInformation)
               .then(() => {
+                mainStore.commit(
+                  'updateUsername',
+                  userBasicInformation.username,
+                );
+                if (userBasicInformation.username) {
+                  userProfile.profile = {
+                    ...userProfile.profile,
+                    username: userBasicInformation.username,
+                  };
```

Comment:
If your intention is to update only the username field, you can do it directly by assigning the property:
```suggestion
                  userProfile.profile.username = userBasicInformation.username;
```

Another approach is to use the `username` property from the main store, which will update automatically this field. However, for now, this approach is fine.

Reply:
Country flag update in user profile

https://github.com/user-attachments/assets/87772fc4-bde9-4abb-a1c3-bdc9dce5c4e3

Reply:
Awesome!!!